doc/conf.py

```diff
@@ -237,6 +237,7 @@
     'Transform': 'mne.transforms.Transform',
     'Coregistration': 'mne.coreg.Coregistration',
     'Figure3D': 'mne.viz.Figure3D',
+    'MNE-object': ':term:`MNE time-series object`',
```

Comment:
I'd rather not do this in this PR. In most places we just do `instance of Raw, Epochs, or Evoked`. It's not optimal and in some cases not even complete (e.g., can be AverageTFR, STC, etc. instead) but it's common and people are used to it. If we want to go this route it should be in a separate PR where we change a ton of these at once. But for now I'd just make this `instance of Raw, Epochs, Evoked, AverageTFR, or SourceEstimate` or so.

But even separate-PR aside, I'm not totally convinced we need to or want to create this new category just for the purposes of making our `Returns` section better. For example, to me the best solution would be to somehow magically infer the class name being documented but I don't think this is possible based on how autodoc works. To me the simplest solution is to do something like `instance of original type` or something more compact that has this same meaning to say "you get back the same instance mostly for chaining purposes".

We should discuss this separately in a new issue. @alexrockhill if you're motivated by this problem, could you try making a new issue with the points that people have made about this already, then we can continue the discussion there?

Reply:
+1 for instance of original type, it is a tricky corner case for autodoc. Sounds good, I'll remove.

Reply:
... but for now in this PR please just use a list and open an issue where we discuss separately what the correct verbiage/linking should be for this "original type" stuff would be